package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,305 @@
+        super(EDRReader, self).__init__(**kwargs)
+
+    def _read_next_step(self) -> EDRStep:
+        """Read next auxiliary step and update ``auxstep``.
+
+        Returns
+        -------
+        AuxStep object
+            Updated with the data for the new step.
+
+        Raises
+        ------
+        StopIteration
+            When end of auxiliary data set is reached.
+        """
+        auxstep = self.auxstep
+        new_step = self.step + 1
+        if new_step < self.n_steps:
+            auxstep._data = {term: self.data_dict[term][self.step + 1]
+                             for term in self.terms}
+            auxstep.step = new_step
+            return auxstep
+        else:
+            self.rewind()
+            raise StopIteration
+
+    def _go_to_step(self, i: int) -> EDRStep:
+        """ Move to and read i-th auxiliary step.
+
+        Parameters
+        ----------
+        i: int
+            Step number (0-indexed) to move to
+
+        Returns
+        -------
+        :class:`EDRStep`
+
+        Raises
+        ------
+        ValueError
+            If step index not in valid range.
+        """
+        if i >= self.n_steps or i < 0:
+            raise ValueError("Step index {0} is not valid for auxiliary "
+                             "(num. steps {1})".format(i, self.n_steps))
+        self.auxstep.step = i - 1
+        self.next()
+        return self.auxstep
+
+    def read_all_times(self) -> np.ndarray:
+        """ Get list of time at each step.
+
+        Returns
+        -------
+        NumPy array of float
+            Time at each step.
+        """
+        return self.data_dict[self.time_selector]
+
+    def get_data(self, data_selector: Union[str, List[str], None] = None) \
+            -> Dict[str, np.ndarray]:
+        """ Returns the auxiliary data contained in the :class:`EDRReader`.
+        Returns either all data or data specified as `data_selector` in form
+        of a str or a list of any of :attr:`EDRReader.terms`. `Time` is
+        always returned to allow easy plotting. """
+        if data_selector is None:
+            return self.data_dict
+
+        elif isinstance(data_selector, list):
+            data_dict = {"Time": self.data_dict["Time"]}
+            for term in data_selector:
+                try:
+                    data_dict[term] = self.data_dict[term]
+                except KeyError:
+                    raise KeyError(f"data_selector {term} invalid. Check "
+                                   "the EDRReader's `terms` attribute.")
+            return data_dict
+
+        else:
+            term = data_selector
+            data_dict = {"Time": self.data_dict["Time"]}
+            try:
+                data_dict[term] = self.data_dict[term]
+            except KeyError:
+                raise KeyError(f"data_selector {term} invalid. Check the "
+                               "EDRReader's `terms` attribute.")
+            return data_dict
```

Comment:
```suggestion
        def _get_data_term(term, datadict):
            try:
                return datadict[term]
            except KeyError:
                raise KeyError(f"data selector {term} is invalid. Check the "
                                       "EDRReader's `terms` attribute.)

        data_dict = {"Time": self.data_dict["Time"]}

        # will fit under 79 chargs if data_selector -> data_keys
        # otherwise just add the if/else back in but each calling _get_data_term
        terms = [data_selector] if not isinstance(data_selector, list) else data_selector

        for term in terms:
            data_dict[term] = _get_data_term(term, self.data_dict)

        return data_dict
```